adapters/Fyber/FyberAdapter/GADMediationAdapterFyber.m

```diff
@@ -1,81 +1,243 @@
+    [self.adSpot fetchAdWithCompletion:^(IAAdSpot * _Nullable adSpot, IAAdModel * _Nullable adModel, NSError * _Nullable error) {
+        __strong typeof(self) strongSelf = weakSelf;
+
+        if (error) {
+            [strongSelf.connector adapter:strongSelf didFailAd:error];
+        } else {
+            [strongSelf.connector adapterDidReceiveInterstitial:strongSelf];
+        }
+    }];
+}
+
+- (void)loadRewardedAdForAdConfiguration:(GADMediationRewardedAdConfiguration *)adConfiguration completionHandler:(GADMediationRewardedLoadCompletionHandler)completionHandler {
+    _rewardedAd = [[GADFYBMediationRewardedAd alloc] init];
+    [self.rewardedAd loadRewardedAdForAdConfiguration:adConfiguration completionHandler:completionHandler];
+}
+
+- (void)stopBeingDelegate {
+    if (self.viewUnitController.unitDelegate) {
+        self.viewUnitController.unitDelegate = nil;
+    }
+    
+    if (self.fullscreenUnitController.unitDelegate) {
+        self.fullscreenUnitController.unitDelegate = nil;
+    }
+}
+
+- (void)presentInterstitialFromRootViewController:(UIViewController *)rootViewController {
+    [self.fullscreenUnitController showAdAnimated:YES completion:nil];
+}
+
+#pragma mark - Service
+
+- (IAAdRequest * _Nonnull)buildRequestWithSpotId:(NSString *)spotId {
+    IAAdRequest *request = [IAAdRequest build:^(id<IAAdRequestBuilder>  _Nonnull builder) {
+        builder.useSecureConnections = NO;
+        builder.spotID = spotId;
+        builder.timeout = 10;
+        builder.keywords = [[self.connector.userKeywords valueForKey:@"description"] componentsJoinedByString:@""];
+        
+        if ([self.connector userHasLocation]) {
+            builder.location = [[CLLocation alloc] initWithLatitude:self.connector.userLatitude longitude:self.connector.userLongitude];
+        }
+    }];
+    
+    return request;
+}
+
+- (void)initBannerWithRequest:(IAAdRequest * _Nonnull)request {
+    _MRAIDContentController = [IAMRAIDContentController build:^(id<IAMRAIDContentControllerBuilder>  _Nonnull builder) {}];
+    
+    _viewUnitController = [IAViewUnitController build:^(id<IAViewUnitControllerBuilder>  _Nonnull builder) {
+        builder.unitDelegate = self;
+        [builder addSupportedContentController:self.MRAIDContentController];
+    }];
+    
+    _adSpot = [IAAdSpot build:^(id<IAAdSpotBuilder>  _Nonnull builder) {
+        builder.adRequest = request;
+        [builder addSupportedUnitController:self.viewUnitController];
+    }];
+}
+
+- (void)initInterstitialWithRequest:(IAAdRequest * _Nonnull)request {
+    _MRAIDContentController = [IAMRAIDContentController build:^(id<IAMRAIDContentControllerBuilder>  _Nonnull builder) {}];
+    _videoContentController = [IAVideoContentController build:^(id<IAVideoContentControllerBuilder>  _Nonnull builder) {}];
+
+    _viewUnitController = [IAViewUnitController build:^(id<IAViewUnitControllerBuilder>  _Nonnull builder) {
```

Comment:
It looks like this is for banners, so unclear why to set this on interstitials. The fullscreenUnitController has both video and MRAID demand.

Reply:
@maninomula You are right, the viewUnitController is redundant here.